### DISCUSSION TITLE
how to assign a midi device?

### DISCUSSION THREAD

**Discussion by drnotube** [RomanKubiak/ctrlr#122] [General]
hi,
I am trying to assign a midi port as follows,
`sendMidiOut = function(--[[ CtrlrModulator --]] mod, --[[ number --]] value, --[[ number --]] source)

panel:setPropertyInt("panelMidiOutputDevice", value)

end`

but I am not successful,
is there any other way?

thanks

**Comment by Ted-juh** [comment]
To set the device:
panel:setPropertyString("panelMidiOutputDevice","DEVICE_NAME")
(DEVICE_NAME) should match the combo input text in midiOut settings of Ctrlr)

To set midichannel:
panel:setPropertyString("panelMidiOutPutChannelDevice", VALUE)

Or first, check what device and channel are used:

midiOutputDevice = panel:getProperty(“panelMidiOutputDevice”)
midiOutputchannel = panel:getProperty(“panelMidiOutputChannelDevice”)

console(“currentOutDevice = “..midiOutputDevice)
console(“currentOutChannel = “..midiOutputChannel)

**Comment by drnotube** [comment]
ok it's a string!
thanks you!
Ctrlr is the best!
I already have the Yamaha DJXII keyboard panel finished, I'm going to upload it to the web.

**Reply by DRBAUS** [reply]
This is what I also need.

panel:setPropertyString("panelMidiOutPutChannelDevice", VALUE) does not work for me.

panel:setProperty("panelMidiOutputChannelDevice", 1, false) does.

Thanks for posting the question and answers!
